server/routes/phase-allocation.js

```diff
@@ -22,25 +23,34 @@ router.get(
     keycloak.getUserInfoMiddleware(),
   ],
   asyncMiddleware(async (req, res) => {
-    const { hcapUserInfo: user } = req;
+    /**
+     * @type {import("../keycloak").hcapUserInfo} hcapUserInfo
+     */
+    const user = req.hcapUserInfo;
     const siteId = parseInt(req.params.id, 10);
 
-    // TODO: HCAP-1336 check that user has access to siteId
-    if (siteId) {
-      const result = await getAllSitePhases(siteId);
+    // Validate request
+    if (!siteId) return res.status(400).send('Invalid site ID');
+    const authorized =
+      user.isSuperUser ||
+      user.isMoH ||
+      ((user.isHA || user.isEmployer) &&
+        (await getSitesForUser(user)).map((site) => site.id).includes(siteId));
+    if (!authorized) return res.status(403).send('Unauthorized site ID');
 
```

Comment:
To clarify - superUser and MOH can see every site? HA and employer are restricted to sites in their health authority region?

Reply:
Yes